### PR TITLE
[SYCL] Fix failing specialization constants tests.

### PR DESCRIPTION
**Pull request by maarquitos14**
Recent changes made to specialization constants tests relied on `--debug-only` option, which is only enabled when assertions are enabled. This commit guards the commands using such an option to be run only when assertions are enabled.

This should fix the post-commit CI failure observed after #12647 was merged.

### CODE REVIEW

File: llvm/test/tools/sycl-post-link/spec-constants/SYCL-2020-zeroinitializer-array-of-arrays.ll [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ; RUN: sycl-post-link --spec-const=native -S < %s -o %t.files.table
 ; RUN: FileCheck %s -input-file=%t.files_0.ll --check-prefix CHECK-IR
 ; RUN: FileCheck %s -input-file=%t.files_0.prop --check-prefix CHECK-PROP
-; RUN: sycl-post-link -debug-only=SpecConst --spec-const=native -S < %s 2>&1 | FileCheck %s --check-prefix=CHECK-LOG
+; RUN: %if asserts %{ sycl-post-link -debug-only=SpecConst --spec-const=native -S < %s 2>&1 | FileCheck %s --check-prefix=CHECK-LOG %} %else %{ %}
```
Comment (bader) [inline llvm/test/tools/sycl-post-link/spec-constants/SYCL-2020-zeroinitializer-array-of-arrays.ll:4]: As we don't test "else" branch of this condition, it's better to put `; REQUIRES: asserts`. LIT will report accurate results in this case.

Comment (maarquitos14) [reply llvm/test/tools/sycl-post-link/spec-constants/SYCL-2020-zeroinitializer-array-of-arrays.ll:4]: As far as I understand, `REQUIRES: asserts` would affect to the whole test (all the `RUN` lines), I only want it to affect the one line.

File: llvm/test/tools/sycl-post-link/spec-constants/composite-default-value-padding.ll [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ; RUN: sycl-post-link -spec-const=emulation < %s -o %t.files.table
 ; RUN: FileCheck %s -input-file=%t.files_0.prop
-; RUN: sycl-post-link -debug-only=SpecConst -spec-const=emulation < %s 2>&1 | FileCheck %s --check-prefix=CHECK-LOG
+; RUN: %if asserts %{ sycl-post-link -debug-only=SpecConst -spec-const=emulation < %s 2>&1 | FileCheck %s --check-prefix=CHECK-LOG %} %else %{ %}
```
Comment (aelovikov-intel) [inline llvm/test/tools/sycl-post-link/spec-constants/composite-default-value-padding.ll:3]: Empty `else` isn't necessary.

**Comment by aelovikov-intel** [comment]
Should be up to the @intel/dpcpp-tools-reviewers  if they're ok with reduced coverage (not performing checks in release builds in testing).

**Comment by maarquitos14** [comment]
> Should be up to the @intel/dpcpp-tools-reviewers if they're ok with reduced coverage (not performing checks in release builds in testing).

Actually, what was added in #12647 was extra coverage that required `--debug-only`, so I added new `RUN` lines for every test with the mentioned option. Conditioning the new `RUN` lines is not reducing the coverage any further than it was before #12647.

**Review by aelovikov-intel — APPROVED** [review]

**Review by maksimsab — APPROVED** [review]
LGTM

**Pull request merged**